types.go

```diff
@@ -125,7 +126,10 @@ func NewMeteredOptionalType(gauge common.MemoryGauge, typ Type) OptionalType {
 func (OptionalType) isType() {}
 
 func (t OptionalType) ID() string {
-	return fmt.Sprintf("%s?", t.Type.ID())
+	if len(t.typeID) == 0 {
```

Comment:
I'm not sure what the convention is for checking empty strings. It seems we have 40 instances of `len\(.*== *0` and 41 instances of `== *""` in the repo, but the first regex also includes arrays/slices.

Reply:
hmm, they're used interchangeably in the go stdlib and produce identical assembly: https://godbolt.org/z/fib1x1

entirely just a style choice, nothing big